APITests/filename_manifest_validate.py

```diff
@@ -0,0 +1,143 @@
+        # openpyxl was best suited to process data of this type
+        workbook = openpyxl.load_workbook(exel_data)
+        manifest_sheet = workbook["Sheet1"]
+
+        # taken from openpyxl documentation on how to process a worksheet with headers and indices into a dataframe
+        data = manifest_sheet.values
+        cols = next(data)[1:]
+        data = list(data)
+        idx = [r[0] for r in data]
+        data = (islice(r, 1, None) for r in data)
+        manifest = pd.DataFrame(data, index=idx, columns=cols)
+        manifest = manifest.dropna(axis=1,how='all').dropna(axis=0,how="all")
+        manifest.reset_index(drop=False,names='Component',inplace=True)
+
+        manifest = self._generate_errors(manifest)
+        self._save_manifest(manifest)
+
+        return
+
+
+    def _generate_errors(self, manifest: pd.DataFrame) -> pd.DataFrame:
+        manifest.iloc[::2,1]=manifest.iloc[::2,1].map(lambda x: f"err {x}")
+        manifest.iloc[1::2,2]=manifest.iloc[1::2,2].map(lambda x: x[:-1:])
+
+        return manifest
+
+    def _save_manifest(self, manifest: pd.DataFrame) -> None:
+        cwd = os.getcwd()
+        self.manifest_path = os.path.join(cwd, f"APITests/test_manifests/filename_{self.num_files}_manifest_{uuid.uuid4()}.csv")
+        manifest.to_csv(self.manifest_path,index=False)
+
+        return
+
+
+    def validate_filename_manifest(self):
+        params = self.params
+        params["data_type"] = "MockFilename"
+
+        dt_string, time_diff, status_code_dict = asyncio.run(
+            cal_time_api_call(
+                url=self.base_url,
+                params=params,
+                headers=self.headers,
+                concurrent_threads=CONCURRENT_THREADS,
+                manifest_to_send_func=self.cal_run_time.send_manifest_async,
+                file_path_manifest=self.manifest_path,
+            )
+        )
+
+        return format_run_time_result(
+            endpoint_name="model/validate",
+            description=f"Validate a mockFilename manifest from the example data model. The manifest has around {self.num_files} rows with a 100% error rate.",
+            data_schema="Example data schema",
+            num_rows=self.num_files,  # number of rows of the manifest being validated
+            data_type=params["data_type"],  # data type
+            restrict_rules=False,  # Restrict rules is set to False
+            dt_string=dt_string,
+            num_concurrent=CONCURRENT_THREADS,
+            latency=time_diff,
+            status_code_dict=status_code_dict,
+        )
+
+def monitor_filename_manifest_validator() -> Row:
+    schema_url = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld"
+    project_sizes=[30,300,3000]
+    rows = []
+    for file_num in project_sizes:
+        mv = FilenameManifestValidate(url=schema_url, num_files=file_num)
+        filename_mv = replace(mv, base_url=DEV_REFACTOR_URL)
```

Comment:
here probably shoudn't hard code it to DEV REFACTOR URL?

Reply:
The `filenameExists` rule does not exist on the AWS dev environment. Since this profiling was the acceptance criteria for a new feature and not just a part of the regular profiling we do every release, it seemed necessary to run this profiling before we released to PyPi and deployed to AWS. If I switch the url to `BASE_URL` then when we run the profiling it will raise an error as the validation rule does not exist in the version currently deployed to the dev environment.
See also comment [above](https://github.com/Sage-Bionetworks/schematic_profiler/pull/50#discussion_r1772290425):
> This profiling requires functionality that is present on dev-refactor and not the dev instance used in BASE_URL. When I inquired about deploying these changes to the dev environment I was told to use dev-refactor instead. After the next release we can switch to using BASE_URL

If you still feel like it would be better to switch now then I can do so @linglp

Reply:
Oh I see. Got it. Thanks for the update. If that's the case, we could just use DEV REFACTOR URL for now. After schematic release, the code can then be updated to use base url :)  Thanks, I will mark this comment as resolved.